src/tests/class_tests/openms/source/NASequence_test.cpp

```diff
@@ -0,0 +1,751 @@
+{
+  // tested via (void setSequence(const std::vector< const Ribonucleotide * > &s))
+  TEST_EQUAL(true, true);
+}
+END_SECTION
+
+START_SECTION((void set(size_t index, const Ribonucleotide* r)))
+{
+  NASequence aaaa = NASequence::fromString("AAAA");
+  NASequence cccc = NASequence::fromString("CCCC");
+  aaaa.set(2, cccc.get(2));
+  TEST_EQUAL(aaaa, NASequence::fromString("AACA"));
+}
+END_SECTION
+
+START_SECTION((const Ribonucleotide* get(size_t index)))
+{
+  // tested via ((void set(size_t index, const Ribonucleotide *r)))
+  TEST_EQUAL(true, true);
+}
+END_SECTION
+
+START_SECTION((const Ribonucleotide*& operator[](size_t index)))
+{
+  NASequence aaa = NASequence::fromString("AAA");
+  const NASequence ggg = NASequence::fromString("GGG");
+  aaa[1] = ggg[2];
+  TEST_EQUAL(aaa, NASequence::fromString("AGA"));
+}
+END_SECTION
+
+START_SECTION((const Ribonucleotide*& operator[](size_t index) const ))
+{
+  NASequence aaa = NASequence::fromString("AAA");
+  const NASequence ggg = NASequence::fromString("GGG");
+  aaa[1] = ggg[2];
+  TEST_EQUAL(aaa[1], ggg[0]);
+}
+END_SECTION
+
+START_SECTION((bool empty() const ))
+{
+  NASequence aaa;
+  TEST_EQUAL(aaa.empty(), true);
+}
+END_SECTION
+
+START_SECTION((size_t size() const ))
+{
+  NASequence seq;
+  TEST_EQUAL(seq.size(), 0);
+  seq = NASequence::fromString("UGG");
+  TEST_EQUAL(seq.size(), 3);
+  // don't count terminal phosphate in sequence length:
+  seq = NASequence::fromString("pUGG");
+  TEST_EQUAL(seq.size(), 3);
+  seq = NASequence::fromString("UGGp");
+  TEST_EQUAL(seq.size(), 3);
+  seq = NASequence::fromString("pUGGp");
+  TEST_EQUAL(seq.size(), 3);
+}
+END_SECTION
+
+START_SECTION((void clear()))
+{
+  NASequence aaa = NASequence::fromString("AAA");
+  aaa.clear();
+  TEST_EQUAL(aaa.empty(), true);
+}
+END_SECTION
+
+START_SECTION((bool hasFivePrimeMod() const ))
+{
+  NASequence aaa = NASequence::fromString("AAA");
+  TEST_EQUAL(aaa.hasFivePrimeMod(), false);
+}
+END_SECTION
+
+START_SECTION((void setFivePrimeMod(const RibonucleotideChainEnd* r)))
+{
+  NASequence aaa = NASequence::fromString("AAA");
+  TEST_EQUAL(aaa.hasFivePrimeMod(), false);
+  aaa.setFivePrimeMod(db->getRibonucleotide("(pN)"));  // 5' phosphate
+  TEST_EQUAL(aaa.hasFivePrimeMod(), true);
+  TEST_EQUAL(aaa.getFivePrimeMod()->getCode(), "(pN)");
+  TEST_STRING_EQUAL(aaa.toString(), "[(pN)]AAA");
+}
+END_SECTION
+
+START_SECTION((const RibonucleotideChainEnd* getFivePrimeMod() const))
+{
+  // tested via (const RibonucleotideChainEnd* getFivePrimeMod() const )
+  TEST_EQUAL(true, true);
+}
+END_SECTION
+
+START_SECTION((void setThreePrimeMod(const RibonucleotideChainEnd* r)))
+{
+  NASequence aaa = NASequence::fromString("AAA");
+  TEST_EQUAL(aaa.hasThreePrimeMod(), false);
+  aaa.setThreePrimeMod(db->getRibonucleotide("(pN)"));
+  TEST_EQUAL(aaa.hasThreePrimeMod(), true);
+  TEST_EQUAL(aaa.getThreePrimeMod()->getCode(), "(pN)");
+  TEST_STRING_EQUAL(aaa.toString(), "AAA[(pN)]");
```

Comment:
can you explain the notation? In peptides we use `[]` to indicate mass differences not mapped to a specific modification. Does it make sense to use the same notation for a different purpose here?